android/runtime/v8/src/native/modules/APIModule.cpp

```diff
@@ -115,7 +115,19 @@ void APIModule::logInfo(const FunctionCallbackInfo<Value>& args)
 	Isolate* isolate = args.GetIsolate();
 	HandleScope scope(isolate);
 	v8::String::Utf8Value message(isolate, APIModule::combineLogMessages(args));
-	APIModule::logInternal(LOG_LEVEL_INFO, LCAT, *message);
+
+	std::string cppStr(*message);
+
+	int maxChunk = 4050;
+	int chunkCount = cppStr.length() / maxChunk;
```

Comment:
You are truncating the decimal when the result is cast to an int. You need to do a `ceil(cppStr.length() / maxChunk)`.

You could also completely do away with the `chunkCount` by incrementing `i` by `maxChunk` each iteration.

Reply:
done :+1: I've retested it with a short log, a 4050 character log and 4050+ logs. All are visible